d2layouts/d2sequence/layout.go

```diff
@@ -141,14 +141,26 @@ func cleanup(g *d2graph.Graph, sequenceDiagrams map[string]*sequenceDiagram, obj
 		obj.LabelPosition = go2.Pointer(string(label.InsideTopCenter))
 		sd := sequenceDiagrams[obj.AbsID()]
 
-		// shift the sequence diagrams as they are always placed at (0, 0)
-		sd.shift(obj.TopLeft)
+		// shift the sequence diagrams as they are always placed at (0, 0) with some padding
+		sd.shift(
+			geo.NewPoint(
+				obj.TopLeft.X+GROUP_CONTAINER_PADDING,
+				obj.TopLeft.Y+GROUP_CONTAINER_PADDING,
+			),
+		)
 
 		obj.Children = make(map[string]*d2graph.Object)
+		obj.ChildrenArray = make([]*d2graph.Object, 0)
 		for _, child := range sd.actors {
 			obj.Children[child.ID] = child
+			obj.ChildrenArray = append(obj.ChildrenArray, child)
+		}
+		for _, child := range sd.groups {
+			if child.Parent == obj {
```

Comment:
something tells me this won't work with TALA as the object references change in that call

Reply:
changed to IDs. that'll work right